index.d.ts

```diff
@@ -108,6 +108,19 @@ export declare interface Tracer extends opentracing.Tracer {
    * should not be cached.
    */
   getRumData(): string;
+
+  /**
+   * Links an authenticated user to the current trace.
+   * @param {object} user Properties of the authenticated user. Accepts custom fields.
+   * @param {string} user.id Unique identifier of the user. Mandatory.
+   * @param {string} [user.email] Email of the user.
+   * @param {string} [user.name] User-friendly name of the user.
+   * @param {string} [user.session_id] Session ID of the user.
+   * @param {string} [user.role]
+   * @param {string} [user.scope]
+   * @returns {Tracer} The Tracer instance for chaining.
+   */
+  setUser(user: object): Tracer;
```

Comment:
This could use a better type.

Reply:
I'm REALLY unfamiliar with TS typings and stuff, you mean having a `@typedef` or something ? how could that look like ?

Reply:
I'm also very unfamiliar with it, but there are other examples in this file of custom types that are then used as parameters. The idea is that we list at the very least the properties that are officially supported.

Reply:
Here i've added an interface declaration like TracerOptions